src/lib.rs

```diff
@@ -1,744 +1,345 @@
-                set_input_topology: lib.get(b"IRCompilerSetInputTopology")?,
-                set_local_root_signature: lib.get(b"IRCompilerSetLocalRootSignature")?,
-                set_minimum_deployment_target: lib.get(b"IRCompilerSetMinimumDeploymentTarget")?,
-                set_minimum_gpu_family: lib.get(b"IRCompilerSetMinimumGPUFamily")?,
-                set_shared_rt_arguments: lib.get(b"IRCompilerSetSharedRTArguments")?,
-                set_stage_in_generation_mode: lib.get(b"IRCompilerSetStageInGenerationMode")?,
-                set_stream_out_enabled: lib.get(b"IRCompilerSetStreamOutEnabled")?,
-                set_strip_cut_index: lib.get(b"IRCompilerSetStripCutIndex")?,
-                set_tessellation_enabled: lib.get(b"IRCompilerSetTessellationEnabled")?,
-                set_tessellator_output_primitive: lib
-                    .get(b"IRCompilerSetTessellatorOutputPrimitive")?,
-                set_validation_flags: lib.get(b"IRCompilerSetValidationFlags")?,
-                set_vertex_render_target_index_id: lib
-                    .get(b"IRCompilerSetVertexRenderTargetIndexID")?,
-                set_vertex_viewport_index_id: lib.get(b"IRCompilerSetVertexViewportIndexID")?,
-            };
+            self.funcs
+                .IRCompilerSetHitgroupType(self.me, hit_group_type)
+        }
+    }
 
-            let me = (funcs.create)();
+    pub fn synthesize_indirect_intersection_function(
+        &mut self,
+        target_metallib: &mut IRMetalLibBinary,
+    ) -> bool {
+        unsafe {
+            self.funcs
+                .IRMetalLibSynthesizeIndirectIntersectionFunction(self.me, target_metallib.me)
+        }
+    }
 
-            Ok(Self { funcs, me })
+    pub fn synthesize_indirect_ray_dispatch_function(
+        &mut self,
+        target_metallib: &mut IRMetalLibBinary,
+    ) -> bool {
+        unsafe {
+            self.funcs
+                .IRMetalLibSynthesizeIndirectRayDispatchFunction(self.me, target_metallib.me)
         }
     }
 
-    pub fn set_global_root_signature(&mut self, root_signature: &IRRootSignature) {
+    pub fn set_entry_point_name(&mut self, new_name: &CStr) {
         unsafe {
-            (self.funcs.set_global_root_signature)(self.me, root_signature.me);
+            self.funcs
+                .IRCompilerSetEntryPointName(self.me, new_name.as_ptr())
         }
     }
 
     pub fn alloc_compile_and_link(
         &mut self,
-        entry_points: &[&[u8]],
-        input: &'lib IRObject,
-    ) -> Result<IRObject<'lib>, Box<dyn std::error::Error>> {
-        let entry_points = entry_points
-            .iter()
-            .map(|e| e.as_ptr())
-            .collect::<Vec<*const u8>>();
-        let mut error: *mut IRErrorOpaque = unsafe { std::ptr::null_mut::<IRErrorOpaque>().add(0) };
+        entry_point: &CStr,
+        input: &IRObject,
+    ) -> Result<IRObject, Box<dyn std::error::Error>> {
+        let mut error: *mut IRError = std::ptr::null_mut::<IRError>();
```

Comment:
Let's simplify this one too:
```suggestion
        let mut error = std::ptr::null_mut();
```